app/src/main/java/com/sdpteam/connectout/authentication/GoogleLoginActivity.java

```diff
@@ -1,56 +1,73 @@
 package com.sdpteam.connectout.authentication;
 
+import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract;
+import com.sdpteam.connectout.R;
+import com.sdpteam.connectout.drawer.DrawerActivity;
+import com.sdpteam.connectout.profile.ProfileFirebaseDataSource;
+import com.sdpteam.connectout.registration.CompleteRegistrationActivity;
+
 import android.content.Intent;
 import android.os.Bundle;
+import android.view.View;
 import android.widget.Button;
-
+import android.widget.ProgressBar;
+import android.widget.TextView;
 import androidx.activity.result.ActivityResultLauncher;
 import androidx.appcompat.app.AppCompatActivity;
 
-import com.firebase.ui.auth.FirebaseAuthUIActivityResultContract;
-import com.sdpteam.connectout.R;
-import com.sdpteam.connectout.registration.CompleteRegistrationActivity;
-
 /**
  * Redirect automatically to next view (CompleteRegistrationActivity) if the user is already authenticated
  */
 public class GoogleLoginActivity extends AppCompatActivity {
-    private Authentication authentication = new GoogleAuth();
+    public static Authentication authentication = new GoogleAuth();
+    private Button loginBtn;
+    private ProgressBar progressBar;
+    private TextView infoText;
     private final ActivityResultLauncher<Intent> fireBaseSignInLauncher = registerForActivityResult(
             new FirebaseAuthUIActivityResultContract(),
             firebaseActivityResult -> {
                 redirectIfAuthenticated();
             });
 
-    void setAuthenticationService(Authentication a) {
-        // useful for mocking in tests
-        this.authentication = a;
-    }
-
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_google_login);
 
-        Button b = findViewById(R.id.loginButton);
-        b.setOnClickListener(v -> {
+        loginBtn = findViewById(R.id.loginButton);
+        loginBtn.setOnClickListener(v -> {
             fireBaseSignInLauncher.launch(authentication.buildIntent());
         });
 
+        progressBar = findViewById(R.id.progressBar);
+        progressBar.setVisibility(View.GONE);
+
+        infoText = findViewById(R.id.infoText);
+        infoText.setVisibility(View.GONE);
+
         redirectIfAuthenticated();
     }
 
     void redirectIfAuthenticated() {
         AuthenticatedUser authenticatedUser = authentication.loggedUser();
         if (authenticatedUser != null) {
+            loginBtn.setVisibility(View.GONE);
+
+            infoText.setVisibility(View.VISIBLE);
+            infoText.setText("Hello " + authenticatedUser.name + "\nyou will be redirected in few seconds");
```

Comment:
I would suggest refactoring this string into a final static string

Reply:
Now I created a string.xml resource entry which is even better